### PR TITLE
Support `SHOW ALL VERBOSE` to show settings description

### LINKED ISSUE #7736: Add settings description in `SHOW ALL` output
### Is your feature request related to a problem or challenge?

I found its required lots of time to get both settings description, current value and name.

The code contains description, initial value and concealed full name through macros.
The runtime shows the current value, and full name, without description

### Describe the solution you'd like

Add settings description in `SHOW ALL` output to collect all the settings runtime value into single place

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

<!--
We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For example `Closes #123` indicates that this PR will close issue #123.
-->

Closes #7736 .

## Rationale for this change
Expand SHOW ALL stmt to show settings description. Currently it requires going through the code to find the description for the specific datafusion setting, moreover code covered by macros which makes harder to understand the full setting name.

<!--
 Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.  
-->

## What changes are included in this PR?
The PR is to add settings description to the `SHOW ALL` output

```
+------------------------------------------------------------+---------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| name                                                       | value                     | description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+------------------------------------------------------------+---------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| datafusion.catalog.create_default_catalog_and_schema       | true                      | Whether the default catalog and schema should be created automatically.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| datafusion.catalog.default_catalog                         | datafusion                | The default catalog name - this impacts what SQL queries use if not specified                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
| datafusion.catalog.default_schema                          | public                    | The default schema name - this impacts what SQL queries use if not specified                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| datafusion.catalog.format                                  |                           | Type of `TableProvider` to use when loading `default` schema                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| datafusion.catalog.has_header                              | false                     | If the file has a header   
```

<!--
There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.
-->

## Are these changes tested?
Yes
<!--
We typically require tests for all PRs in order to:
1. Prevent the code from being accidentally broken by subsequent changes
2. Serve as another way to document the expected behavior of the code

If tests are not included in your PR, please explain why (for example, are they covered by existing tests)?
-->

## Are there any user-facing changes?
Yes, the `SHOW ALL` output have extra `description` column and confusing `settings` column name was renamed to `value`
<!--
If there are user-facing changes then we may require documentation to be updated before approving the PR.
-->

<!--
If there are any breaking changes to public APIs, please add the `api change` label.
-->

### CODE REVIEW

File: datafusion/sqllogictest/test_files/information_schema.slt [hunk 1]
```diff
@@ -136,91 +136,91 @@ statement ok
...
-datafusion.optimizer.repartition_file_scans true
-datafusion.optimizer.repartition_joins true
-datafusion.optimizer.repartition_sorts true
-datafusion.optimizer.repartition_windows true
-datafusion.optimizer.skip_failed_rules false
-datafusion.optimizer.top_down_join_key_reordering true
-datafusion.sql_parser.dialect generic
-datafusion.sql_parser.enable_ident_normalization true
-datafusion.sql_parser.parse_float_as_decimal false
+datafusion.catalog.create_default_catalog_and_schema true Whether the default catalog and schema should be created automatically.
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/information_schema.slt:142]: My slight concern is the output now may be too wide, and we prob should also consider to split `SHOW ALL`

`SHOW ALL EXTENDED` or `SHOW ALL VERBOSE` will output the description whereas `SHOW ALL` can still show  just name and value

**Review by alamb — COMMENTED** [review]
Thank you @comphead  -- I really like that the descriptions are now included.

I think the change to make it possible to select all the settings directly from `df_settings` is great 💯 

```
select * from information_schema.df_settings
```

As you mention, I am not so sure about including the description in `SHOW <name>` and `SHOW ALL` as it makes the output much more verbose. 

I think it would be a nicer experience if `SHOW ALL` kept the current behavior and we extended `SHOW ALL VERBOSE` and `SHOW <value> VERBOSE` that added the description too.

I checked and this appears to be supported by the parser so we could probably make this work without waiting on an upstream change

```
❯ show all verbose;
0 rows in set. Query took 0.002 seconds.
```

What do you think?

**Comment by comphead** [comment]
@alamb thanks. My concern with the first sentence as the developer who introducing the new param is not aware about concise first sentence requirements and this will still eventually lead to the wide output. 

Having another verbose pair makes much more sense for me.
```
SHOW ALL VERBOSE
SHOW value VERBOSE
```

**Comment by alamb** [comment]
> Having another verbose pair makes much more sense for me.

I agree

**Comment by comphead** [comment]
> > Having another verbose pair makes much more sense for me.
> 
> I agree

Added `SHOW ALL VERBOSE`, `SHOW param VERBOSE` support. 
User doc updated

File: datafusion/sqllogictest/test_files/information_schema.slt [hunk 2]
```diff
@@ -202,12 +202,85 @@ datafusion.sql_parser.dialect generic
 datafusion.sql_parser.enable_ident_normalization true
 datafusion.sql_parser.parse_float_as_decimal false
 
+# show all variables with verbose
+query TTT rowsort
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/information_schema.slt:206]: 🥇

File: docs/source/user-guide/cli.md [hunk 3]
```diff
@@ -397,11 +397,13 @@ Available commands inside DataFusion CLI are:
...
 - Show configuration options
 
+`SHOW ALL [VERBOSE]`
+
 ```SQL
 > show all;
 
 +-------------------------------------------------+---------+
-| name                                            | setting |
+| name                                            | value   |
```
Comment (alamb) [inline docs/source/user-guide/cli.md:406]: I like this change in column name 👍

**Review by alamb — APPROVED** [review]
Looks really nice to me -- thank you @comphead

**Comment by alamb** [comment]
cc @waitingkuo

**Comment by alamb** [comment]
Thanks again @comphead -- this is a really nice contribution

**Pull request merged**